STCommon/src/commonMain/kotlin/com/infomaniak/multiplatform_swisstransfer/common/interfaces/transfers/File.kt

```diff
@@ -31,5 +32,55 @@ interface File {
     val receivedSizeInBytes: Long
     val path: String?
     val thumbnailPath: String?
-}
+    val parent: File?
+    val children: List<File>
+
+    /**
+     * Recursively searches for a child with the given name.
+     *
+     * @param fileName The name of the child to search for.
+     * @return The [File] object representing the found child, or null if not found.
+     */
+    fun findChildByName(fileName: String): File? {
+        if (this.fileName == fileName) return this
+
+        children.forEach { child ->
+            child.findChildByName(fileName)?.let {
+                return it
+            }
+        }
+
+        return null
+    }
```

Comment:
Make it as recursive tail

Reply:
Finally, I think it will become unnecessary with realm query in order to get better performance.